CMakeLists.txt

```diff
@@ -19,7 +19,11 @@ set_property(GLOBAL PROPERTY USE_FOLDERS ON)
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
 
 find_package(YCM)
-find_package(YARP 3.7 REQUIRED COMPONENTS os sig dev math rosmsg idl_tools)
+find_package(YARP 3.7 REQUIRED COMPONENTS os sig dev math idl_tools)
+find_package(YARP_rosmsg)
+if(rosmsg_FOUND)
```

Comment:
Are you sure that this works? I would expect this variable should be named `YARP_rosmsg_FOUND` ?

Reply:
You are absolutely right. I probably didn't test enough (or at all) the situation where the YARP_rosmsg component is present...